src/connection.c

```diff
@@ -808,18 +808,35 @@ static noPollConnOpts * createConnOpts (char * extra_headers, bool secure)
 	return opts;   
 }
 
-
+static void close_conn ( noPollConn *conn, const char *specified_reason)
+{
+	const char *effective_reason = specified_reason;
+	
+	if (NULL == effective_reason) {
+      nopoll_conn_close_ext(conn, CloseNoStatus, NULL, 0);
+      return;
+    }
+    if (strcmp (effective_reason, "SIGUSR2") == 0) {
+		char *sigusr2_reason;
+        if (readFromFile (get_parodus_cfg()->close_reason_file, &sigusr2_reason)) {
+          nopoll_conn_close_ext(conn, CloseNormalClosure, sigusr2_reason, 
+			strlen (sigusr2_reason));
+		  ParodusInfo ("Closed by SIGUSR2, reason: %s\n", sigusr2_reason);
+		  free (sigusr2_reason);
+		  return;
+		}
+		/* file could not be read. use canned message. */
+        effective_reason = "restarted with SIGUSR2";
```

Comment:
@schmidtw what hardcoded default reason would you prefer for the SIGUSR2 restart case?
These are the current values:
parodus_stopping - default, system_restarting - SIGUSR1